### PR TITLE
✨ Adapters

### PR DESCRIPTION
**Pull request by axlon**
I'm sorry about the size of this PR, but it's primarily old files being deleted. I will add back the tests and each driver in separate PRs to not make this one any bigger

### CODE REVIEW

File: src/Contracts/UnsupportedOperationException.php [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace LaraCrafts\UrlShortener\Contracts;
```
Comment (nerg4l) [inline src/Contracts/UnsupportedOperationException.php]: I am not sure putting an exception in the contracts namespace.

File: src/Builder.php [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+    /**
+     * Create a new shortening builder instance.
+     *
+     * @param \LaraCrafts\UrlShortener\Contracts\Client $client
+     * @param \Psr\Http\Message\UriInterface|string $uri
+     * @param array $options
+     * @return void
+     */
+    public function __construct(Client $client, $uri, array $options)
```
Comment (nerg4l) [inline src/Builder.php]: As it is a builder I would not put options as a parameter, at least not as a required one.

File: src/Contracts/UnsupportedOperationException.php (hunk 1 above)
Comment (axlon) [reply src/Contracts/UnsupportedOperationException.php]: Where would you put this then? I kinda just copied this from Laravel hahaha

Comment (nerg4l) [reply src/Contracts/UnsupportedOperationException.php]: Simply into `LaraCrafts\UrlShortener\Exceptions` namespace.

Comment (axlon) [reply src/Contracts/UnsupportedOperationException.php]: For now I'll leave it, if we need to add any more exceptions, they'll get their own namespace

**Pull request merged**